### ISSUE TITLE
The Table component should have a required prop that renders the <caption> element

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#218]
The `<Table/>` component should have a required prop that accepts a string and renders the HTML `<caption>` element. To preserver the current UX, the element could be for screen readers only.

## Possible Implementation

```jsx
<Table caption="My Favorite Table"></Table>
```

Would render to...
```html
<table>
  <caption class="screen-reader-only">My Favorite Table</caption>
</table>
```

**Comment by jonambas** [comment]
Will be addressed in UX-213
